### PR TITLE
manifest: Update the Matter SDK version to 1.1.0.1 TAG

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Update the Matter SDK version to 1.1.0.1 TAG to prepare the sdk_connectedhomeip version for NCS 2.4.0 release.

- [x] Re-generate ZAP files
- [x] Update samples (no differences)
- [x] Update Matter-related links in https://github.com/nrfconnect/sdk-nrf/blob/main/doc/nrf/links.txt
- [x] Update release notes
- [x] Update tables in https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/index.html and https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/protocols/matter/overview/dev_model.html#ug-matter-overview-dev-model-support

### CODE REVIEW

File: doc/nrf/protocols/matter/overview/dev_model.rst [hunk 1]
```diff
@@ -36,11 +36,14 @@ Supported Matter versions in the |NCS|
...
-| Matter version  | Overview of changes                          | Release date        |
-+=================+==============================================+=====================+
-| 1.0.0           | Initial version of the Matter specification. | November 2, 2022    |
-+-----------------+----------------------------------------------+---------------------+
++-----------------+------------------------------------------------------------------------------------------------------------------+---------------------+
+| Matter version  | Overview of changes                                                                                              | Release date        |
++=================+==================================================================================================================+=====================+
+| 1.1.0           | - Improved ICD support: more :ref:`ug_matter_configuring_optional_persistent_subscriptions`, improved monitoring | Unreleased          |
+|                 | - Overall enhancements and bugfixes for Matter Specification, Certification Test Plan, and Matter SDK            |                     |
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/overview/dev_model.rst:43]:
```suggestion
|                 | - Enhancements and bug fixes for Matter Specification, Certification Test Plan, and the Matter SDK.            |                     |
```

Comment (greg-fer) [inline doc/nrf/protocols/matter/overview/dev_model.rst:43]:
```suggestion
|                 | - Enhancements and bug fixes for Matter Specification, Certification Test Plan, and the Matter SDK.            |                     |
```

File: doc/nrf/protocols/matter/overview/dev_model.rst [hunk 2]
```diff
@@ -36,11 +36,14 @@ Supported Matter versions in the |NCS|
...
 
-+-----------------+----------------------------------------------+---------------------+
-| Matter version  | Overview of changes                          | Release date        |
-+=================+==============================================+=====================+
-| 1.0.0           | Initial version of the Matter specification. | November 2, 2022    |
-+-----------------+----------------------------------------------+---------------------+
++-----------------+------------------------------------------------------------------------------------------------------------------+---------------------+
+| Matter version  | Overview of changes                                                                                              | Release date        |
++=================+==================================================================================================================+=====================+
+| 1.1.0           | - Improved ICD support: more :ref:`ug_matter_configuring_optional_persistent_subscriptions`, improved monitoring | Unreleased          |
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/overview/dev_model.rst:42]:
```suggestion
| 1.1.0           | - Improved ICD support: more :ref:`ug_matter_configuring_optional_persistent_subscriptions`, improved monitoring. | Unreleased          |
```
Can you expand ICD acronym to <name (ICD)>? "improved monitoring" of what?

File: doc/nrf/protocols/matter/index.rst [hunk 3]
```diff
@@ -22,6 +22,8 @@ The |NCS| allows you to develop applications with different versions of Matter,
 +--------------------------+-----------------------------------------------------+------------------------+
 | nRF Connect SDK version  | Matter specification version                        | `Matter SDK version`_  |
 +==========================+=====================================================+========================+
+| v2.3.99 (latest)         | :ref:`1.1.0 <ug_matter_overview_dev_model_support>` | 1.1.0.1                |
```
Comment (greg-fer) [inline doc/nrf/protocols/matter/index.rst:25]: Looks good!

**Review by greg-fer — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -117,6 +117,7 @@ The following list summarizes the most important changes inherited from the upst
   Generated onboarding codes should be put on the device's package or on the device itself.
   For details, see the Generating onboarding codes section on the :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.
 * Introduced ``SLEEPY_ACTIVE_THRESHOLD`` parameter that makes the Matter sleepy device stay awake for a specified amount of time after network activity.
+* Added device finish and device color attributes to the Basic Information cluster and added the proper entries in factory data set.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:120]:
```suggestion
* Updated the Basic Information cluster with device finish and device color attributes and added the related entries in factory data set.
```

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/protocols/matter/overview/dev_model.rst [hunk 5]
```diff
@@ -36,11 +36,14 @@ Supported Matter versions in the |NCS|
...
-+=================+==============================================+=====================+
-| 1.0.0           | Initial version of the Matter specification. | November 2, 2022    |
-+-----------------+----------------------------------------------+---------------------+
++-----------------+--------------------------------------------------------------------------------------------------------------------------------+---------------------+
+| Matter version  | Overview of changes                                                                                                            | Release date        |
++=================+================================================================================================================================+=====================+
+| 1.1.0           | - Improved Intermittently Connected Device (ICD) support: more :ref:`ug_matter_configuring_optional_persistent_subscriptions`. | Unreleased          |
+|                 | - Enhancements and bug fixes for Matter Specification, Certification Test Plan, and the Matter SDK.                            |                     |                     |
```
Comment (LuDuda) [inline doc/nrf/protocols/matter/overview/dev_model.rst:43]:
```suggestion
|                 | - Enhancements and bug fixes for Matter Specification, Certification Test Plan, and the Matter SDK.                            |                     |
```

**Review by LuDuda — APPROVED** [review]

**Review by kkasperczyk-no — COMMENTED** [review]
As far as I remember there are some differences regarding usage of generate.py script to generate source files. Could you make sure that guide in `adding_clusters.rst` is up to date and if not align it?

**Comment by kkasperczyk-no** [comment]
> As far as I remember there are some differences regarding usage of `generate.py` script to generate source files. Could you make sure that guide in `adding_clusters.rst` is up to date and if not align it?

If you want to benefit from green CI we can also leave it as is and do a follow-up :laughing: But please in the end make sure the guide instructions are correct.

**Review by kkasperczyk-no — APPROVED** [review]

**Pull request merged**